app/views/supermarkets/show.html.erb

```diff
@@ -0,0 +1,7 @@
+<h1><%= @supermarket.name %></h1>
+
+<h3>Customers that have shopped here:</h3>
+
+<% @customers.each do |customer| %>
```

Comment:
Here is where I would just call your model method directly on `@supermarket` and get rid of the instance variable in your controller.